packages/mdc-menu/simple/foundation.js

```diff
@@ -275,6 +273,27 @@ class MDCSimpleMenuFoundation extends MDCFoundation {
     }
   }
 
+  /**
+   * Handle clicks and cancel the menu if not a list item
+   * @param {!Event} evt
+   * @private
+   */
+  handleDocumentClick_(evt) {
+    let el = evt.target;
+
+    while (el && el !== document.body) {
```

Comment:
Would it make sense to refactor this logic out of `handleDocumentClick_` and into `eventTargetHasClass`? Maybe renaming it to something like `eventTargetOrAncestorHasClass`.

That would make the logic easier to reuse and simpler to test.

Reply:
It would also make the adapter method harder to write and I think we're all for simple adapter methods... so it's up to which is higher priority.

If we ever end up tackling #1104 this might be a candidate for that...

Reply:
Yeah, I think we've always tried to keep the adapter methods as close to one line as possible.